### ISSUE TITLE
Phonegap dont build

### ISSUE THREAD

**Issue by aoreinoso** [j3k0/cordova-plugin-purchase#862]
```
Phone Gap 8.0.0
Plugin Version: 8.1.0
<preference name="phonegap-version" value="cli-8.0.0" />
```
```
<plugin name="cordova-plugin-purchase" spec="^8.1.0">
    <variable name="BILLING_KEY" value="MIIB.....AB" />
</plugin>
```
Error: /var/gimlet/tmp/47458146212820/3009243/gradlew: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/var/gimlet/tmp/47458146212820/3009243/app/src/main/java/com/smartmobilesoftware/util/IabHelper.java:33: error: package com.android.vending.billing does not exist
import com.android.vending.billing.IInAppBillingService;
                                  ^
/var/gimlet/tmp/47458146212820/3009243/app/src/main/java/com/smartmobilesoftware/util/IabHelper.java:102: error: cannot find symbol
    IInAppBillingService mService;
    ^
  symbol:   class IInAppBillingService
  location: class IabHelper
/var/gimlet/tmp/47458146212820/3009243/app/src/main/java/com/smartmobilesoftware/util/IabHelper.java:234: error: package IInAppBillingService does not exist
                mService = IInAppBillingService.Stub.asInterface(service);
                                               ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 errors

**Comment by j3k0** [comment]
PhoneGap CLI 8.0.0 ships with cordova-android@7.0.0, is this correct?

https://build.phonegap.com/current-support

Can you send the output of command ` ionic info`?

**Comment by aoreinoso** [comment]
What can I do?
with other versions of phonegap cli or cordova-plugin-purchase  does not work either

**Comment by j3k0** [comment]
Can you confirm the version of cordova-android you have installed? Maybe run `cordova info` in the root directory (or `phonegap info`?) sorry I do not use phonegap, I don't know the tool so well.

**Comment by aoreinoso** [comment]
I use https://build.phonegap.com
and yes PhoneGap CLI 8.0.0 ships with cordova-android@7.0.0 


pd: I dont speak english.. i did not understand "I bugged, no need for ionic info ;-) there should be an equivalent command with phonegap"

**Comment by j3k0** [comment]
@aoreinoso I guess it's not the answer you want to hear, but it's supposed to work ;-)

I'll try to install phonegap on my computer and see what's happening.

In the meantime, you could try to use the plugin at version `7.2.5`, might work (let me know if it does).

**Comment by aoreinoso** [comment]
...
Fetching plugin "cordova-plugin-purchase@^7.2.5" via npm
Installing "cordova-plugin-purchase" at "7.2.8" for android
...
because: 
Failed to fetch plugin cordova-plugin-purchase@7.2.5 via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Error: version not found: cordova-plugin-purchase@7.2.5
...
and the mistake is the same:

:app:compileDebugJavaWithJavac/var/gimlet/tmp/47458146212300/3009243/app/src/main/java/com/smartmobilesoftware/util/IabHelper.java:33: error: package com.android.vending.billing does not exist
import com.android.vending.billing.IInAppBillingService;
                                  ^
/var/gimlet/tmp/47458146212300/3009243/app/src/main/java/com/smartmobilesoftware/util/IabHelper.java:102: error: cannot find symbol
    IInAppBillingService mService;
    ^
  symbol:   class IInAppBillingService
  location: class IabHelper
/var/gimlet/tmp/47458146212300/3009243/app/src/main/java/com/smartmobilesoftware/util/IabHelper.java:234: error: package IInAppBillingService does not exist
                mService = IInAppBillingService.Stub.asInterface(service);
                                               ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 errors
 FAILED

**Comment by j3k0** [comment]
I didn't own ID `cordova-plugin-purchase` at version `7.2.5`. You need to change to `cc.fovea.cordova.purchase`.

```xml
<plugin name="cc.fovea.cordova.purchase" spec="^7.2.5">
    <variable name="BILLING_KEY" value="MIIB.....AB" />
</plugin>
```

**Comment by aoreinoso** [comment]
The same error
...
Fetching plugin "cc.fovea.cordova.purchase@^7.2.5" via npm
Installing "cc.fovea.cordova.purchase" at "7.4.3" for android
...

**Comment by j3k0** [comment]
OK... I'm missing something.

**Comment by j3k0** [comment]
See https://www.npmjs.com/package/cc.fovea.cordova.purchase --- version `7.2.5` exists there. Why wouldn't phonegap accept to install it?

**Comment by j3k0** [comment]
Can you share your full `config.xml`?

**Comment by aoreinoso** [comment]
I have made an app only with cordova-plugin-purchase and the error is the same ... 

if you want to try it
I have used: config.xml
```

	<?xml version="1.0" encoding="UTF-8" ?>
	<widget xmlns="http://www.w3.org/ns/widgets" xmlns:gap="http://phonegap.com/ns/1.0" xmlns:cdv="http://cordova.apache.org/ns/1.0" xmlns:android="http://schemas.android.com/apk/res/android" android-versionCode="1" version="0.0.1" id="com.ssss.hhhh"  >
	<!-- This file was generated by the Intel XDK Cordova Package Build tool. -->
	<name>MMMMMMMMM</name>
	<description>mmm</description>
	<author>vvvv</author>
	<content src="index.html"/>
	<preference name="android-build-tool" value="gradle" />
	<platform name="android" >
		<allow-intent href="http://*/*"/>
		<allow-intent href="https://*/*"/>
		<allow-intent href="tel:*"/>
		<allow-intent href="sms:*"/>
		<allow-intent href="mailto:*"/>
		<allow-intent href="geo:*"/>
		<allow-intent href="market:*"/>
		<access origin="*"/>
	</platform>
	<!-- https://build.phonegap.com/current-support -->
	<preference name="phonegap-version" value="cli-8.0.0" />
	<preference name="xwalkMultipleApk" value="false" />
	<plugin name="cordova-plugin-purchase">
		<variable name="BILLING_KEY" value="MIIBIj......AB" />
	</plugin>

	<platform  name="android">
		<preference name="SplashMaintainAspectRatio" value="true"/>
	</platform>
	<preference  name="xwalkCommandLine" value="--disable-pull-to-refresh-effect --ignore-gpu-blacklist"/>
	<platform name="android" >
		<preference name="android-installLocation" value="auto"/>
		<preference name="Fullscreen" value="false"/>
		<preference name="Orientation" value="portrait"/>
		<preference name="android-signed" value="true"/>
		<preference name="AndroidLaunchMode" value="singleTop"/>
	</platform>
	<config-file platform="android" parent="/manifest" mode="merge">
		<uses-feature android:name="android.hardware.camera" android:required="false" />
		<uses-feature android:name="android.hardware.camera.autofocus" android:required="false" />
		<uses-feature android:name="android.hardware.camera.flash" android:required="false" />
		<uses-permission android:name="android.permission.CAMERA" />
		<uses-permission android:name="ANDROID.PERMISSION.READ_EXTERNAL_STORAGE" />
		<uses-permission android:name="android.permission.WAKE_LOCK" />
		<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
		<uses-permission android:name="com.android.vending.BILLING" />
	</config-file>
	<config-file platform="android" parent="/manifest/application" mode="merge"> 
		<activity android:launchMode="singleTop" />
	</config-file>
	</widget>
```

and my file hierarchy is:
.
├── config.xml
├── icon.png
├── LICENSE.md
├── package-assets
├── README.md
├── screenshot.png
└── www
    ├── css
    ├── index.html
    ├── js
    └── lib

**Comment by j3k0** [comment]
I see some places you have to specify `source="npm"`, does it help? (I don't have phonegap on this computer to test)

```xml
<plugin name="cc.fovea.cordova.purchase" spec="^7.2.5" source="npm">
```

**Comment by aoreinoso** [comment]
the same result


pd: I do not have it installed either ... you create your zip with the sources, you register in phonegap and upload it...

**Comment by aoreinoso** [comment]
I have tried with the same result (from https://www.npmjs.com/package/cc.fovea.cordova.purchase/v/7.2.5)
```
<gap:plugin name="cc.fovea.cordova.purchase" source="npm" version="6.0.0">
   <param name="BILLING_KEY" value="MIIB...."/>
</gap:plugin>
```

**Comment by Dexus** [comment]
Please install the latest version and re-add android and iOS platform if else nothing helps!

Von meinem iPhone gesendet

**Comment by aoreinoso** [comment]
I have found
https://github.com/AlexDisler/cordova-plugin-inapppurchase/pull/207
I think I need version 7.1.4 of Android in Cordova.

**Comment by aoreinoso** [comment]
it works with (for example):
<preference name="phonegap-version" value="cli-7.1.0" />
and
<gap:plugin name="cc.fovea.cordova.purchase" source="npm" version="6.0.0">
   <param name="BILLING_KEY" value="MIIBIj....B"/>
</gap:plugin>

**Comment by aoreinoso** [comment]
<preference name="phonegap-version" value="cli-7.1.0" />
...
Fetching plugin "cordova-plugin-purchase" via npm
Installing "cordova-plugin-purchase" at "8.1.0" for android
...
then error:
at com.smartmobilesoftware.inappbilling.InAppBillingPlugin.getPublicKey(InAppBillingPlugin.java:156)
---------------
it works with
```
<preference name="phonegap-version" value="cli-7.1.0" />
```
and
<gap:plugin name="cc.fovea.cordova.purchase" source="npm" version="6.0.0">
   <param name="BILLING_KEY" value="MIIB..."/>
</gap:plugin>

it's not goot but it works

**Comment by j3k0** [comment]
Yes, problem is `phonegap-cli` still ships with a cordova-android that had an issue with installing `.aidl` files, hopefully you can upgrade all to latest when they upgrade to `cordova-android@8`

**Comment by aoreinoso** [comment]
Thank you very very much for your time

**Comment by j3k0** [comment]
Welcome.

For `phonegap cli-8.0.0`, you can now use this:

```xml
<plugin spec="https://github.com/j3k0/cordova-plugin-purchase.git#phonegap-cli-8.0.0">
    <param name="BILLING_KEY" value="MIIB..."/>
</plugin>
```

It corresponds to version `8.1.0` of the plugin, with a fix so it builds with `phonegap-cli-8.0.0`.

I couldn't make a working version for `phonegap-cli-8.1.1` (I tried hard), because it ships with `cordova-android@7.1.2` which has a bug with installing `.aidl` files on Android.
